word_guess_lv.rb

```diff
@@ -0,0 +1,312 @@
+
+# We color the program in certain messages and use features like blink and bold in them also.
+require 'colorize'
+
+
+puts "WELCOME TO DISNEY WORD GUESS"
+puts "Words Theme: Walt Disney".colorize(:blue)
+
+# one class contain different methods for the game
+class WordGuess
+
+  def initialize
+    @words_low = %w[minnie mickey daisy donald pluto goofy anna elsa alice]
+    @words_medium = %w[tigger cinderella bambi pocahontas eeyor rabbit mulan belle dumbo]
+    @words_high = %w[esmeralda quasimodo anastasia mowgli anastasia drizella kocoum maleficent]
+    @letters_guessed = []
+    @guesses_counter = 0
+    selection_level
+    @word = @word_array.sample
+    @letters_correct = Array.new(@word.length, "_" )
+    ascii_image
+    letters_input
+  end
+
+# This method will select the level of the game and according to the level, the program
+# will select an arrays of words.
+# The user will have 4 trials to enter a level. We did this to have an end of the program
+# in case the user keeps entering the wrong level input.
+  def selection_level
+
+    levels = %w[low medium high]
+    puts "Please enter a level: low, medium or high?"
+    level = nil
+    until level == "low" || level == "medium" || level == "high"
+      level = gets.chomp.downcase
+      if level == "low"
+        @word_array = @words_low
+      elsif level == "medium"
+        @word_array = @words_medium
+      elsif level == "high"
+        @word_array = @words_high
+      elsif level != "low" && level != "hard" && level != "medium"
+        puts "Invalid level. Please enter again"
+        level = gets.chomp.downcase
+      end
+
+      attempt = 0
+      while !levels.include?(level)
+        if attempt < 3
+          attempt += 1
+          puts "Please enter a valid level. You have #{(-attempt + 3) + 1} more attempts.\n"
+          level = gets.chomp
+        else
+          puts "Too many attempts..start the program again"
+          exit
+        end
+      end
+    end
+
+  end
+
+# This method hanbles inappropriate user input.
+# This method validates that the input of letter from the user is a letter and not any other character.
+# We did this to don't count inputs as wrong guesses if the user inputs numbers, symbols etc.
+  def validate_input_letter(input)
+    until !input.match(/[^a-z]/)
+      puts "\nPlease return a valid letters from A to Z.\n"
```

Comment:
This loop conditional is a little confusing, since you have three negations: `until` (backwards `while`), the `!`, and the `^` in the regex.

Instead, you might say `while input.match(/[^a-z]/)`.